Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -1678,55 +1681,105 @@ void HTMLMediaElement::loadResource(const URL& initialURL, const ContentType& co
 
     updateVolume();
 
-    bool loadAttempted = false;
-#if ENABLE(MEDIA_SOURCE)
-    if (!m_mediaSource && url.protocolIs(mediaSourceBlobProtocol) && !m_remotePlaybackConfiguration) {
-        if (RefPtr mediaSource = MediaSource::lookup(url.string()))
-            m_mediaSource = MediaSourceInterfaceMainThread::create(mediaSource.releaseNonNull());
+    auto contentType = initialContentType;
+
+    if (m_blob && !m_remotePlaybackConfiguration) {
+        ALWAYS_LOG(LOGIDENTIFIER, "loading generic blob");
+        if (!m_blobURLForReading.isEmpty())
+            ThreadableBlobRegistry::unregisterBlobURL(m_blobURLForReading);
+        m_blobURLForReading = { BlobURL::createPublicURL(document().protectedSecurityOrigin().ptr()), document().topOrigin().data() };
+        ThreadableBlobRegistry::registerBlobURL(document().protectedSecurityOrigin().ptr(), document().policyContainer(), m_blobURLForReading, m_blob->url());
+
+        url = m_blobURLForReading;
+        if (contentType.isEmpty())
+            contentType = ContentType { m_blob->type() };
     }
 
-    if (m_mediaSource) {
-        loadAttempted = true;
+    auto completionHandler = [url, player = m_player, keySystem, this](const ContentType& contentType) {
```

Comment:
Capturing `this` here without any kind of protection is fragile, see my other comment.

Reply:
I'll make it so that it doesn't capture this, but instead receive HTMLMediaElement& as argument, that would alleviate your concerns I think

Reply:
strike the above.